src/understand-tidb/mvcc-garbage-collection.md

```diff
@@ -1 +1,280 @@
++--------+------------+----------------------------------------+----------------------------------------+
+|     58 |          1 | 7480000000000000385f698000000000000001 | 7480000000000000385f698000000000000002 |
++--------+------------+----------------------------------------+----------------------------------------+
+1 row in set (0.002 sec)
+
+MySQL [test]> SELECT HIGH_PRIORITY job_id, element_id, start_key, end_key FROM mysql.gc_delete_range_done WHERE 1;
++--------+------------+--------------------+--------------------+
+| job_id | element_id | start_key          | end_key            |
++--------+------------+--------------------+--------------------+
+|     55 |         53 | 748000000000000035 | 748000000000000036 |
++--------+------------+--------------------+--------------------+
+1 row in set (0.002 sec)
+```
+
+The to-be-deleted ranges are stored in `mysql.gc_delete_range`, in GC, they will be deleted in `deleteRanges`. After they are cleaned up, they will be moved into `mysql.gc_delete_range_done`, and double-checked after 24 hours.
+
+Finally, the `GCWorker` is going to clean up the stale keys. There are `central` mode and `distributed` mode. From TiDB 3.0, `tikv_gc_mode` is set to `distributed` mode by default, this document will talk about `distributed` mode only.
+
+Distributed GC is implemented by push up the safepoint in PD. Notice that the safepoint is monotonic, PD guarantees it by comparing the old and new values [here](https://github.com/tikv/pd/blob/v5.2.1/server/grpc_service.go#L1081-L1127).
+
+```go
+// UpdateGCSafePoint implements gRPC PDServer.
+func (s *Server) UpdateGCSafePoint(ctx context.Context, request *pdpb.UpdateGCSafePointRequest) (*pdpb.UpdateGCSafePointResponse, error) {
+	...
+	newSafePoint := request.SafePoint
+
+	// Only save the safe point if it's greater than the previous one
+	if newSafePoint > oldSafePoint {
+		if err := s.storage.SaveGCSafePoint(newSafePoint); err != nil {
+			return nil, err
+		}
+		log.Info("updated gc safe point",
+			zap.Uint64("safe-point", newSafePoint))
+	} else if newSafePoint < oldSafePoint {
+		log.Warn("trying to update gc safe point",
+			zap.Uint64("old-safe-point", oldSafePoint),
+			zap.Uint64("new-safe-point", newSafePoint))
+		newSafePoint = oldSafePoint
+	}
+
+	return &pdpb.UpdateGCSafePointResponse{
+		Header:       s.header(),
+		NewSafePoint: newSafePoint,
+	}, nil
+}
+```
+
+If PD returns the same new safepoint as TiDB provides, TiDB takes this GC round success.
+
+## TiKV Part
+
+This is a document about TiDB dev guide, so there will simply introduce how TiKV works with TiDB with less implementations.
+
+### Distributed GC
+
+As we talked above, TiDB only push up the safepoint in PD instead of cleaning up keys directly. Every TiKV has an inner [`GcManager`](https://github.com/tikv/tikv/blob/v5.2.1/src/server/gc_worker/gc_manager.rs#L217-L237) with `safe_point` field.
```

Comment:
```suggestion
As we talked above, TiDB only pushes up the safepoint in PD instead of cleaning up keys directly. Every TiKV has an inner [`GcManager`](https://github.com/tikv/tikv/blob/v5.2.1/src/server/gc_worker/gc_manager.rs#L217-L237) with `safe_point` field.
```